modules/drivers/oracle/test/metabase/driver/oracle_test.clj

```diff
@@ -461,3 +462,41 @@
   (testing "Oracle should strip double quotes and null characters from identifiers"
     (is (= "ABC_D_E__FG_H"
            (driver/escape-alias :oracle "ABC\"D\"E\"\u0000FG\u0000H")))))
+
+(deftest read-timestamp-with-tz-test
+  (mt/test-driver :oracle
+    (testing "We should return TIMESTAMP WITH TIME ZONE columns correctly"
+      (let [test-date "2024-12-20 16:05:00 US/Eastern"
+            sql       (format "SELECT TIMESTAMP '%s' AS timestamp_tz FROM dual" test-date)
+            query     (-> (mt/native-query {:query sql})
+                          (assoc-in [:middleware :format-rows?] false))]
+        (doseq [report-tz ["UTC" "US/Pacific"]]
+          (mt/with-temporary-setting-values [report-timezone report-tz]
+            (mt/with-native-query-testing-context query
+              (testing "The value should come back from driver with original zone info, regardless of report timezone"
+                (= (t/offset-date-time (u.date/parse test-date) "US/Eastern")
+                   (ffirst (mt/rows (qp/process-query query))))))))))))
+
+(deftest read-timestamp-with-local-tz-test
+  (mt/test-driver :oracle
+    (testing "We should return TIMESTAMP WITH LOCAL TIME ZONE columns correctly"
+      (let [test-date  "2024-12-20 16:05:00 US/Pacific"
+            sql        (format "SELECT CAST(TIMESTAMP '%s' AS TIMESTAMP WITH LOCAL TIME ZONE) AS timestamp_tz FROM dual" test-date)
+            query      (-> (mt/native-query {:query sql})
+                           (assoc-in [:middleware :format-rows?] false))
+            spec       #(sql-jdbc.conn/db->pooled-connection-spec (mt/db))
+            old-format (-> (jdbc/query (spec) "SELECT value FROM NLS_SESSION_PARAMETERS WHERE PARAMETER = 'NLS_TIMESTAMP_TZ_FORMAT'")
+                           ffirst
+                           val)
+            do-with-temp-tz-format (fn [thunk]
+                                     ;; verify that the value is independent of NLS_TIMESTAMP_TZ_FORMAT
+                                     (jdbc/execute! (spec) "ALTER SESSION SET NLS_TIMESTAMP_TZ_FORMAT = 'YYYY-MM-DD HH:MI'")
+                                     (thunk)
+                                     (jdbc/execute! (spec) (format "ALTER SESSION SET NLS_TIMESTAMP_TZ_FORMAT = '%s'" old-format)))]
```

Comment:
I was a little concerned that parsing the timestamp as a string would be dependent on this setting, but it doesn't seem to be the case. I've added this to the test in case that changes in the future. Here are [the docs](https://docs.oracle.com/en/database/oracle/oracle-database/21/refrn/NLS_TIMESTAMP_TZ_FORMAT.html) for the setting.